subsys/net/ip/icmpv6.c

```diff
@@ -145,8 +145,8 @@ static int icmpv6_handle_echo_request(struct net_icmp_ctx *ctx,
 	 * that it is set properly using a value from neighbor cache.
 	 * Same for source as it points to original pkt ll src address.
 	 */
-	net_pkt_lladdr_dst(reply)->addr = NULL;
-	net_pkt_lladdr_src(reply)->addr = NULL;
+	memset(net_pkt_lladdr_dst(reply)->addr, 0, sizeof(net_pkt_lladdr_dst(reply)->addr));
```

Comment:
I wonder, shouldn't we set `len` to 0 too, as now neigbor cache check seems to rely on this field?

Reply:
Yes, definitely we need to set the len to 0, otherwise the address is considered valid. Will fix.